addons/sale/models/sale_order.py

```diff
@@ -793,6 +793,14 @@ def _unlink_except_draft_or_cancel(self):
                     "You can not delete a sent quotation or a confirmed sales order."
                     " You must first cancel it."))
 
+    def write(self, vals):
+        res = super().write(vals)
+        if 'partner_id' in vals and vals.get('partner_id'):
```

Comment:
Isn't it twice about the same check? I'd rather have
```suggestion
        if vals.get('partner_id', False):
```

Reply:
The `False` is not even necessary `get` will return None if the key is not found, so it's also falsy.